awswrangler/s3/_merge_upsert_table.py

```diff
@@ -0,0 +1,122 @@
+"""Amazon Merge Upsert Module which can upsert existing table (PRIVATE)."""
+
+import logging
+from typing import List, Optional
+
+import boto3
+import pandas
+
+import awswrangler as wr
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _update_existing_table(
+    existing_df: pandas.DataFrame,
+    delta_df: pandas.DataFrame,
+    primary_key: List[str],
+    database: str,
+    table: str,
+    boto3_session: Optional[boto3.Session],
+) -> None:
+    """Perform Update else Insert onto an existing Glue table."""
+    # Set Index on the pandas dataframe so that join/concat can be made
+    existing_df = existing_df.set_index(keys=primary_key, drop=False, verify_integrity=True)
+    delta_df = delta_df.set_index(keys=primary_key, drop=False, verify_integrity=True)
+    # Merge-Upsert the data for both of the dataframe
+    merged_df = pandas.concat([existing_df[~existing_df.index.isin(delta_df.index)], delta_df])
+    # Remove the index and drop the index columns
+    merged_df = merged_df.reset_index(drop=True)
+    # Get existing tables location
+    path = wr.catalog.get_table_location(database=database, table=table)
```

Comment:
Oh you are missing to pass the `boto3_session` argument forward here in the `get_table_location()` call.